BUILDING.md

```diff
@@ -0,0 +1,276 @@
+a personal tree of installed software in `$HOME/opt`, you can inform
+CMake that this path should be searched for packages by passing it as
+a command line "define" to CMake:
+
+``` shell
+cmake -DCMAKE_PREFIX_PATH=$HOME/opt ...
+
+```
+
+Please see
+[CMAKE_PREFIX_PATH](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html)
+for more details on `CMAKE_PREFIX_PATH`.
+
+### Options for Building the SDK
+
+### `CMAKE_INSTALL_PREFIX`
+
+By default, the SDK will install into a subdirectory of the working
+build directory. This is not appropriate for making the SDK available
+for other projects. To install the SDK to a location of your choice
+you should customize `CMAKE_INSTALL_PREFIX` while invoking `CMake` and
+then re-run the `install` task. For instance, to install the SDK to
+`/usr/local`:
+
+
+``` shell
+cmake -DCMAKE_INSTALL_PREFIX=/usr/local ...
+ninja install
+```
+
+#### `VIAMCPPSDK_ENFORCE_COMPILER_MINIMA`
+
+By default, the SDK build will enforce compiler minima. If you must
+bypass this check for some reason, you may do so by disabling this
+default-on option:
+
+``` shell
+cmake -DVIAMCPPSDK_ENFORCE_COMPILER_MINIMA=OFF ...
+```
+
+#### `VIAMCPPSDK_USE_DYNAMIC_PROTOS`
+
+By default, the SDK build will use checked-in copies of generated
+sources. However, these sources only work with a narrow range of gRPC
+versions. If the checked-in copies do not work, or if you would prefer
+to always work against regenerated code, you can request that proto
+generation happen as part of the build by enabling this flag:
+
+``` shell
+cmake -DVIAMCPPSDK_USE_DYNAMIC_PROTOS=ON ...
+```
+
+#### `VIAMCPPSDK_OFFLINE_PROTO_GENERATION`
+
+By default, the SDK will use buf.build services and definitions to
+generate code, which requires online access. If you wish to do
+entirely local proto generation, you may request this:
+
+```shell
+cmake -DVIAMCPPSDK_OFFLINE_PROTO_GENERATION=ON ...
+```
+
+Note that this option only has an effect when
+`VIAMCPPSDK_USE_DYNAMIC_PROTOS=ON`.
+
+## Building Against the SDK
+
+### Using CMake
+
+The Viam C++ SDK installs support files for CMake's `find_package`
+subsystem.
+
+``` cmake
+find_package(viam-cpp-sdk CONFIG REQUIRED COMPONENTS Lib)
+target_link_libraries(mytarget viam-cpp-sdk::viamcpp)
+
+```
+
+Note that you will need to ensure that `CMAKE_PREFIX_PATH` is set
+appropriately in your project to include the path to the installed
+Viam C++ SDK.
+
+An example project using this technique is available under
+`examples/project/cmake` and it can be built with CMake in the regular
+way. If you have allowed the SDK to install to the default
+`build/install` prefix, the example should build against your SDK
+installation automatically. If you have customized the SDK
+installation prefix by building it with a specific
+`CMAKE_INSTALL_PREFIX` that is not among CMake's default package
+search list (e.g. `$HOME/opt`), you will need to pass that information
+along when building the example:
+
+``` shell
```

Comment:
might be helpful if the CMAKE_PREFIX_PATH were being set in this code sample. i find people are quick to miss things that they cant copy/paste or is buried in paragraphs

Reply:
oh wait a second. it does that via the flag below... perhaps not. i failed to install to the correct dir.. nevermind

Reply:
Hmm. My use of `$HOME/opt` was as an example of a "reasonable" place one *might* have installed the SDK that wouldn't be in CMake's default search list. It wasn't really intended as a "you should install it to there" sort of thing. I will think about how to clarify that.

Reply:
I've replaced `$HOME/opt` in the shell example with `<path/to/sdk/install/location>` which should make it clearer that we don't mean `$HOME/opt` in fact.